src/main/java/com/google/sps/placeGuide/PlaceGuide.java

```diff
@@ -0,0 +1,162 @@
+package com.google.sps.placeGuide;
+
+import org.jetbrains.annotations.Nullable;
+import com.google.appengine.api.datastore.GeoPt;
+
+/** Class containing place guide's information. */
+public class PlaceGuide {
+
+  // Unique identifier for a {@code PlaceGuide} automatically given 
+  // when a {@code PlaceGuide} entity is created and will be used to delete
+  // or edit {@code PlaceGuide}.
+  @Nullable
```

Comment:
when can this be null?

Reply:
it will be null only when the user creates a new place guide. the data of the placeguide will be fetched from the client and will be made a new place guide without an id. then when the place guide is being stored in the database, the id will be generated.

Reply:
Null ID is weird, can be confusing and error prone.

To create a place guide: 
1/ Client sends HttpRequest that contains place guide
2/ Servlet receive the place guide creation request:
   2a/ Generate ID
   2b/ Create PlaceGuide using the ID + data from HttpRequest

So ID doesn't need to be null

Reply:
Created place guide id on servlet using datastore so that DatastorePlaceGuideRepository can access the entity based on the created id instead of creating a new id.

Reply:
remove Nullable?